pegen/pegen.c

```diff
@@ -529,8 +529,22 @@ run_parser(struct tok_state *tok, void *(start_rule_func)(Parser *), int mode,
     PyErr_Clear();
 
     p->start_rule_func = start_rule_func;
+    p->error_marker = PyThread_tss_alloc();
```

Comment:
Now this is part of the Parser struct I don't think it needs to be thread-safe also -- the Parser struct is not shared between threads (a new one is created for each parse() call).